docs/analytics-ga4/ga4-ecommerce-tracker.md

```diff
@@ -0,0 +1,122 @@
+
+- on page load
+- the application of filters and/or sorting
+- clicking a search result
+
+## Basic use
+
+In `analytics-ga4.js`:
+
+```JavaScript
+//= require ./analytics-ga4/ga4-ecommerce-tracker
+```
+
+In `live_search.js` in the `finder-frontend` repository (https://github.com/alphagov/finder-frontend/blob/main/app/assets/javascripts/live_search.js#L129):
+
+```JavaScript
+GOVUK.analyticsGa4.Ga4EcommerceTracker.init(isNewPage) // isNewPage is a boolean and is used to detect whether a new page has been loaded or not
+```
+
+Unlike the other GA4 tracking scripts in this repository, ecommerce tracking is not initialised by `init-ga4.js`. Instead, it is initialised by the JS that runs on finder pages (specifically, the `GOVUK.Ga4EcommerceTracking()` function). It is called:
+
+On page load - https://github.com/alphagov/finder-frontend/blob/main/app/assets/javascripts/live_search.js#L55
+
+And whenever search results are updated - https://github.com/alphagov/finder-frontend/blob/main/app/assets/javascripts/live_search.js#L121
+
+Note: ecommerce tracking will only run if there is an element which has the `data-ga4-ecommerce` attribute. If the script can't find an element with this attribute, it will stop running.
+
+## Ecommerce tracking
+
+### On page load
+
+Upon initialisation, the ecommerce tracker script will first detect whether a new page has been loaded (this is determined by the boolean passed when `Ga4EcommerceTracker.init()` is called). If `true`, this indicates that a new page has been loaded (i.e. the user has conducted a new search or has navigated to a new page using the pagination) and the initial set of search results that is loaded is pushed to the `dataLayer`.
+
+For example, if a user searches for "tax", the following object will be pushed to the `dataLayer` on page load:
+
+```JSON
+{
+    "event": "search_results",
+    "search_results": {
+        "event_name": "view_item_list",
+        "results": 65426,
+        "sort": "Relevance",
+        "term": "tax",
+        "ecommerce": {
+            "items": [
+                {
+                    "index": 1,
+                    "item_id": "/vehicle-tax",
+                    "item_list_name": "Search"
+                },
+                {
+                    "index": 2,
+                    "item_id": "/income-tax",
+                    "item_list_name": "Search"
+                },
+                {
+                    "index": 3,
+                    "item_id": "/income-tax/overview",
+                    "item_list_name": "Search"
+                },
+                "etc..."
+            ]
+        }
+    }
+}
+```
+
+For further information on the object structure/schema, please see the implementation guide: https://docs.google.com/document/d/1LGLRb6zMvJBjzMYPgLAcBWbS-GsKGUzLcumCsZgiUhM
```

Comment:
This isn't a public document, so I don't think it should be linked here

Reply:
Good point, have removed. Thank you!